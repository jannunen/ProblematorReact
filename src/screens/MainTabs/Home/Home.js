import * as React from 'react';
import {
  View,
  Text,
  StyleSheet,
  AsyncStorage,
  Dimensions,
} from 'react-native'
import { TabView, TabBar, SceneMap } from 'react-native-tab-view';

import { connect } from 'react-redux';

import { goToAuth } from '../../navigation'
import { USER_KEY } from '../../../../config'
import { selectGym } from '../../../store/actions/index';
import ProblemList from '../../../components/ProblemList/ProblemList';


export class Home extends React.Component {
  handleItemClicked = (problemid) => {
    // Do something
    console.log(problemid);
  }
  firstRoute = () => {
    return (
      <View style={[styles.scene, { backgroundColor: '#252623' }]} >
        <ProblemList handleItemClicked={this.handleItemClicked} />
      </View>
    );
  };
  secondRoute = () => (
    <View style={[styles.scene, { backgroundColor: '#252623' }]} >
    <Text>T4est2</Text>
    </View>
  );
  state = {
    index: 0,
    routes: [
      { key: 'first', title: 'List' },
      { key: 'second', title: 'Map' },
    ],
    problemsLoaded : false
  };

  static get options() {
    return {
      topBar: {
        title: {
          text: 'Problems'
        },
      }
    };
  }
  logout = async () => {
    this.setState({
      code : null
    })
    try {
      await AsyncStorage.removeItem(USER_KEY)
      goToAuth()
    } catch (err) {
      console.log('error signing out...: ', err)
    }
  }

  render() {
    return (
        <TabView
        navigationState={this.state}
        renderScene={SceneMap({
          first: () => {return this.firstRoute(this.props)},
          second: this.secondRoute,
        })}
        onIndexChange={index => this.setState({ index })}
        initialLayout={{ width: Dimensions.get('window').width }}
      />
    )
  }
}
/*
        <Button
          onPress={() => {
            Navigation.push(this.props.componentId, {
              component: {
                name: 'com.padadise.OtherScreen',
              }
            });
          }}
          title="View next screen"
        />

        */

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center'
  },
  scene: {
    flex: 1,
  },

})

const mapStateToProps = (state) => {
  return {
    loading : state.problems.loading,
    error : state.problems.error
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    onSelectGym : gymid => dispatch(selectGym(gymid))
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Home);