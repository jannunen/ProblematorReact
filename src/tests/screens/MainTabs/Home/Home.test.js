import 'react-native';
import React from 'react';
import sinon from 'sinon'
import configureMockStore from 'redux-mock-store'
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import thunk from 'redux-thunk'
import { Home } from '../../../../screens/MainTabs/Home/Home';

const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)


let  history, wrapper;

test('should correctly render LoginPage', () => {
  const store = mockStore({ loading: false, error :false })
  const tree = shallow(<Home />).dive();
  expect(tree).toMatchSnapshot();
});

/*
test('should call onProblemLoad in componentDidMount', () => {
  wrapper = shallow(<Home  {...props}/>)
  wrapper.instance().componentDidMount()
  expect(onLoadProblems.calledOnce).toBe(true)
});
*/

