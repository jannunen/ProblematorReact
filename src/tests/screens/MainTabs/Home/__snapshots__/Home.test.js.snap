// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render LoginPage 1`] = `
<View
  collapsable={false}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <TabBar
    getAccessibilityLabel={[Function]}
    getAccessible={[Function]}
    getLabelText={[Function]}
    getTestID={[Function]}
    jumpTo={[Function]}
    layout={
      Object {
        "measured": false,
        "width": 750,
      }
    }
    navigationState={
      Object {
        "index": 0,
        "problemsLoaded": false,
        "routes": Array [
          Object {
            "key": "first",
            "title": "List",
          },
          Object {
            "key": "second",
            "title": "Map",
          },
        ],
      }
    }
    offsetX={0}
    panX={0}
    position={-0}
    useNativeDriver={false}
  />
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <PagerScroll
      canJumpToTab={[Function]}
      getTestID={[Function]}
      jumpTo={[Function]}
      layout={
        Object {
          "measured": false,
          "width": 750,
        }
      }
      navigationState={
        Object {
          "index": 0,
          "problemsLoaded": false,
          "routes": Array [
            Object {
              "key": "first",
              "title": "List",
            },
            Object {
              "key": "second",
              "title": "Map",
            },
          ],
        }
      }
      offsetX={0}
      panX={0}
      position={-0}
      useNativeDriver={false}
    >
      <SceneComponent
        component={[Function]}
        jumpTo={[Function]}
        key="first"
        route={
          Object {
            "key": "first",
            "title": "List",
          }
        }
      />
      <View
        key="second"
      />
    </PagerScroll>
  </View>
</View>
`;
